### PR TITLE
Add autosubmit code owners

### PR DESCRIPTION
**Pull request by CaseyHillers**
Follow up from [Formalize Cocoon Contributions](https://docs.google.com/document/d/1SxYP41p3CZY_BrUNb_zSfufvhUiSN3qxGl7QEjjTqHk/edit#heading=h.lgp3erb6u85n)

### CODE REVIEW

**Comment by ricardoamador** [comment]
There is already a top level codeowners file. Do we want to just use that one.

**Comment by CaseyHillers** [comment]
> There is already a top level codeowners file. Do we want to just use that one.

Good point. I'm used to how we do it in Google which is next to the closest subdirectory.

@keyonghan do you have opinions?

**Comment by keyonghan** [comment]
I would use the top level OWNERSHIP file to keep it consistent with our other GitHub repos.
As for app_dart, shall we define the ownership on API level?

**Comment by CaseyHillers** [comment]
Thanks for the context, done! Updated to use top level owners

File: CODEOWNERS
Comment (ricardoamador) [inline CODEOWNERS:10]: Can we just use the infra team for this instead of people directly?

Comment (CaseyHillers) [reply CODEOWNERS:10]: Happy to! What group is it? I don't see it in https://github.com/orgs/flutter/teams

Comment (keyonghan) [reply CODEOWNERS:10]: i am a bit reluctant to add the team here. The ownership here will trigger review request for any changes happened from the Dir.
@ricardoamador Are you okay to own the whole auto_submit dir? If not, let's divide it into smaller pieces. Same for the app_dart.
The reason I put the ownership on dir level for cipd packages is they are relatively a small self-contained process. As for app_dart, it would be too large to give a single ownership entry on the dir level. But as for `auto_submit`, up to you @ricardoamador

Comment (ricardoamador) [reply CODEOWNERS:10]: I will defer to Keyong.

Comment (keyonghan) [reply CODEOWNERS:10]: As for auto_submit, I prefer to keep only @ricardoamador (assuming okay:-) as the owner considering he has been the main contact/contributor of this app and the app is not that big. 
As for app_dart, let's skip it for now. We want to break it into smaller pieces on API/file level.

Comment (CaseyHillers) [reply CODEOWNERS:10]: I would go the opposite way. Can we add the entire engprod team as owners? Having a single person be the owner creates a bottleneck, and the best practice is to ensure owners have redundancy.

Comment (keyonghan) [reply CODEOWNERS:10]: CODEOWNERS file here is behaving differently from Google. It just auto adds the owner as a code reviewer, and will not be a bottleneck. People can add anybody else as a code reviewer.

Comment (CaseyHillers) [reply CODEOWNERS:7]: Done. Applied feedback for now to only add the autosubmit CODEOWNERS

**Review by ricardoamador — APPROVED** [review]
LGTM

**Pull request merged**
